src/app/public/modules/modal/modal-adapter.service.ts

```diff
@@ -11,24 +12,39 @@ export class SkyModalAdapterService {
 
   private docRef: any;
   private bodyEl: HTMLElement;
+  private renderer: Renderer2;
+  private hostRef: ComponentRef<SkyModalHostComponent>;
 
   constructor(
-    private windowRef: SkyWindowRefService) {
+    private rendererFactory: RendererFactory2,
+    private windowRef: SkyWindowRefService
+  ) {
+      this.renderer = this.rendererFactory.createRenderer(undefined, undefined);
       this.docRef = this.windowRef.getWindow().document;
       this.bodyEl = this.windowRef.getWindow().document.body;
   }
 
   public addHostEl(): void {
-    this.bodyEl.appendChild(this.docRef.createElement('sky-modal-host'));
+    const domElem = (this.hostRef.hostView as EmbeddedViewRef<any>).rootNodes[0];
```

Comment:
It's a bit strange to be messing with "Angular things" in the DOM adapter (such as embedded views). If you can't get around the breaking-change for this public method, maybe you could take some ideas from Flyout's implementation?

https://github.com/blackbaud/skyux-flyout/blob/master/src/app/public/modules/flyout/flyout.service.ts#L87-L108

If not, maybe the adapter should only know about the `domElem` (and not the embedded view), so do something like:

```
public set hostElement(value: any) {
  this._hostElement = value;
}
```

Which would let you set the element from the modal service:
```
this.adapter.hostElement = (componentRef.hostView as EmbeddedViewRef<any>).rootNodes[0];
```

Reply:
The flyout is what I was basing things on but the flyout version had the dom element passed into the function. I can't do this due to that being a breaking change. I've modified things so that the dom element is now set on the adapter instead of the component ref and so the adapter only does dom related things.